### PR TITLE
SCI-1030 vcf to plink

### PR DESCRIPTION
**Pull request by slzarate**

### CODE REVIEW

**Comment by slzarate** [comment]
Developer

#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [ ] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [x] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [ ] add new input if applicable

#src/code.sh src/code.py
- [ ] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [ ] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [ ] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [x] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.



Optional - not doing for now
- (Optional) Internally, we no longer use the resource project (DX_RESOURCES_ID) for new applets. It's a bit harder to share or bring to other cloud/region. Also, the owner has to maintain the project for the user to use applets. However, it seems that almost half of these applets are extensively using it, so we might have to keep it. I need to think a bit more if DX_RESOURCES_ID has other concern.
- (Optional) Azure support. I note it here because the manuscript will also mention Azure cloud. We have several customers actively use Azure cloud including St.Jude and Stanford. To do this, we need applets and their resources to support Azure. Regarding DX_RESOURCES_ID, we can make a mirror of resource project in Azure with exact md5. Then we just add logic to applets to detect region. 
- (Optional) Add citation to publication after publish.

**Review by nishill — APPROVED** [review]

**Review by nishill — COMMENTED** [review]
change "set -x" to "set -e -x -o pipefail"

File: sequencing/vcf_to_plink/dxapp.json [hunk 1]
```diff
@@ -83,22 +92,46 @@
...
+      "systemRequirements": {
+        "*": {
+          "instanceType": "mem1_ssd1_x4"
+        }
+      }
+    }
+  },
+  "details": {
+    "whatsNew": "*1.0.0: Cleaned up, added documentation, moved to Ubuntu 14.04;",
+    "upstreamAuthor": "Ritchie, MD",
```
Comment (etal) [inline sequencing/vcf_to_plink/dxapp.json]: Should the field `upstreamAuthor` be dropped? The upstream tool is PLINK 1.9, and those authors aren't associated with this app's development. Ritchie lab did not develop PLINK.

File: sequencing/vcf_to_plink/dxapp.json [hunk 2]
```diff
@@ -83,22 +92,46 @@
...
+    "release": "14.04",
+    "distribution": "Ubuntu",
+    "timeoutPolicy": {
+      "main": {
+        "hours": 24
+      }
+    }
+  },
+  "regionalOptions": {
+    "aws:us-east-1": {
```
Comment (etal) [inline sequencing/vcf_to_plink/dxapp.json:114]: Add the Azure region and asset, too?

File: sequencing/vcf_to_plink/dxapp.json (hunk 1 above)
Comment (slzarate) [reply sequencing/vcf_to_plink/dxapp.json]: You're right; I'll remove this line.

File: sequencing/vcf_to_plink/dxapp.json [hunk 3]
```diff
@@ -83,22 +92,60 @@
...
+      }
+    }
+  },
+  "details": {
+    "whatsNew": "*1.0.0: Cleaned up, added documentation, moved to Ubuntu 14.04;",
+    "citations": []
   },
-  "resources" : "project-BYpFk1Q0pB0xzQY8ZxgJFv1V",
   "openSource" : true,
   "developers" : ["user-tnperson", "user-auv13", "user-atf3"],
```
Comment (yihchii) [inline sequencing/vcf_to_plink/dxapp.json:150]: Please review developers and authorizedUsers from dxapp.json

File: sequencing/vcf_to_plink/dxapp.json [hunk 4]
```diff
@@ -83,22 +92,60 @@
...
+      ],
+      "systemRequirements": {
+        "*": {
+          "instanceType": "azure:mem2_ssd1_x4"
+        }
+      }
+    }
+  },
+  "details": {
+    "whatsNew": "*1.0.0: Cleaned up, added documentation, moved to Ubuntu 14.04;",
```
Comment (yihchii) [inline sequencing/vcf_to_plink/dxapp.json:146]: "whatsNew": "\* 1.0.0: Cleaned up, added documentation, moved to Ubuntu 14.04;"
->
"whatsNew": "\* 1.0.0 First version of the Convert VCF file to PLINK file app."

File: sequencing/vcf_to_plink/dxapp.json [hunk 5]
```diff
@@ -83,22 +92,60 @@
...
+      }
+    },
+    "azure:westus": {
+      "assetDepends": [
+        {
+          "project": "project-FGp4k7j9k2Vy68736j2kBVPy",
+          "folder": "/",
+          "version": "0.0.1",
+          "name": "marylyn_asset"
+        }
```
Comment (yihchii) [inline sequencing/vcf_to_plink/dxapp.json:136]: Please update record IDs to:
aws:us-east-1: record-FJ3zf8j08zVZFzqYG88Pbk7z
azure:westus: record-FJ4058j9Vf9ZBBVB2Q7b3VQz

**Review by yihchii — COMMENTED** [review]
Some comments:
1. Please update to use the latest version of the app.
2. Please consider the generalized whatsNew log for the first app release.
3. Please remove authorizedUsers and developers from dxapp.json

File: sequencing/vcf_to_plink/dxapp.json [hunk 6]
```diff
@@ -12,6 +12,7 @@
     {
       "name": "vcf_fn",
```
Comment (yihchii) [inline sequencing/vcf_to_plink/dxapp.json:13]: "vcf_fn" -> "variants_vcfgzs"

Also need to update in the app script accordingly.

File: sequencing/vcf_to_plink/dxapp.json [hunk 7]
```diff
@@ -21,6 +22,7 @@
     {
       "name": "vcfidx_fn",
```
Comment (yihchii) [inline sequencing/vcf_to_plink/dxapp.json:23]: "vcfidx_fn"-> "variants_vcfgztbis"

Also need to update in the app script accordingly.

**Review by yihchii — COMMENTED** [review]
Please update variable names to follow the nomenclature.

**Pull request merged**